src/actions/assetsActions.js

```diff
@@ -371,6 +385,85 @@ export const fetchAccountAssetsBalancesAction = (account: Account) => {
   };
 };
 
+export const fetchAllAccountsTotalsAction = () => {
+  return async (dispatch: Dispatch, getState: GetState) => {
+    if (getState().totals.isFetching) return;
+
+    dispatch({ type: SET_FETCHING_TOTALS, payload: true });
+
+    const accounts = accountsSelector(getState());
+    const smartWalletAccounts = accounts.filter(isNotKeyBasedType);
+
+    smartWalletAccounts.forEach((account) => dispatch(fetchCollectiblesAction(account)));
+
+    const accountsAddresses = smartWalletAccounts.map((account) => getAccountAddress(account));
+    const availableChainProtocols = await getZapperAvailableChainProtocols(accountsAddresses);
+    if (!availableChainProtocols) {
+      reportErrorLog('fetchAllAccountsTotalsAction failed: no availableChainProtocols', { accountsAddresses });
+      return;
+    }
+
+    try {
+      await Promise.all(availableChainProtocols.map(async ({ chain, zapperProtocolIds, zapperNetworkId }) => {
+        const chainProtocolsBalances = await Promise.all(zapperProtocolIds.map(async (zapperProtocolId) => {
+          const protocolBalancesByAccounts = await getZapperProtocolBalanceOnNetwork(
```

Comment:
Yes mate, we're aware, thanks for concerns! This will essentially move either to our serverless or Etherspot back-end, but given time window we have until private beta this is the only solution.